code-challenges.test.js

```diff
@@ -15,48 +20,117 @@
 // --------------------1) Create a function that takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.
 
 // a) Create a test with expect statements using the variables provided.
-
+describe("wordToNum", () => {
 var secretCodeWord1 = "Lackadaisical"
-// Expected output: "L4ck4d41s1c4l"
 var secretCodeWord2 = "Gobbledygook"
-// Expected output: "G0bbl3dyg00k"
 var secretCodeWord3 = "Eccentric"
-// Expected output: "3cc3ntr1c"
-
+it("takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.", () => {
+    expect(wordToNum(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+    expect(wordToNum(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+    expect(wordToNum(secretCodeWord3)).toEqual("3cc3ntr1c")
+  })
+})
+// // Expected output: "L4ck4d41s1c4l"
 
+// Expected output: "G0bbl3dyg00k"
 
-// b) Create the function that makes the test pass.
+// Expected output: "3cc3ntr1c"
 
 
 
+// b) Create the function that makes the test pass.
+// create a function taking in a string as an array
+//iterate 
+//conditionals to see if letters match
+//.replace letters with number.
+//I have gone through slack and used the recources to try and get every letter to change but i was only able to get the letter a to change to 4. No other letters would switch..... I tried changing it into an array
+// I tried many different ways. Im sorry i deleted them insted of commenting out but i left single examples commented(imagine they are on all)
+// const wordToNum = (string) => {
+    // let newString = string.split("")
+    // newString.map(value => {
+    //     if (value === 'a'|| 'A'){
+    //         return value.replace(/[aA]/g, 4)
+    //     }
+    //     return newString.join("")
+    // })
```

Comment:
So this should work if you just tweak one thing. You need to assign new String the value of going through the map. 


```javascript 
 const wordToNum = (string) => {
    let newString = string.split("").map(value => {
         if (value === 'a'|| 'A'){
            return value.replace(/[aA]/g, 4)
        }
         return newString.join("")
     })
    }
``` 
    
If you checked that result you'd end up with the "a's" being replaced. Than just bring in the other letters. 
    
```javascript 
 const wordToNum = (string) => {
    let newString = string.split("").map(value => {
         if (value === 'a'|| 'A'){
            return value.replace(/[aA]/g, 4)
        } else if (value === 'e'|| 'E'){
            return value.replace(/[eE]/g, 3)
        } else if (value === 'i'|| 'I'){
            return value.replace(/[iI]/g, 1)
        } else if (value === 'o'|| 'O'){
            return value.replace(/[oO]/g, 0)
        } else {
        return value
        }
         return newString.join("")
     })
    }
```